broadinstitute/consent
# [DUOS-226][risk=low] Migrate DARs from mongo to postgres

## Addresses
https://broadinstitute.atlassian.net/browse/DUOS-226

See https://broadinstitute.atlassian.net/browse/DUOS-604 for cleanup work

## Changes
* One-off groovy scripts to execute migration
* Copy all DARs from mongo to postgres
* Temporary admin endpoints to facilitate migration
* Read all DAR info from postgres now, aside from migration apis
* Use reference id instead of id as much as possible. Remaining usages will be dealt with when we address the partial dar collection.
* Re-implement some native mongo things such as create date, sorting, date conversions, etc. Sprinkled throughout the codebase
* More tests for service and dao calls.

## Tests
The following workflows are manually verified.

* Export dars from mongo -> import to postgres using groovy script
* Researcher
  * View dar console 
  * Create new dar 
  * Submit new dar 
  * Create partial dar and save 
  * Submit partial dar 
  * Review dar 
  * Cancel dar 
* Admin
  * View dar manage page 
  * Create new dar election 
  * Create migrated dar election 
  * Cancel dar election 
  * Create canceled dar election 
  * View dar summary 
  * View dar election 
  * After chair final vote, election shows in reviewed cases 
* Admin - Statistics
  * DUL: All Cases 
  * DUL: Reviewed Cases 
  * DAR: Should data be granted to this applicant?
    * All Cases 
    * Reviewed Cases 
  * DAR: Was the research purpose accurately converted to a structured format? 
    * All Cases 
    * Reviewed Cases 
  * DAR: Evaluation of automated matching in comparison with the Data Access Committee decision 
    * Cases reviewed by automated matching  
    * Agreement between automated matching and Data Access Committee 
* Member
  * View dar page (filtered by DAC) 
  * View dar after new election created 
  * Vote on migrated dar 
  * Vote on new dar 
  * DAR page shows correct votes 
  * Edit vote 
* Chair
  * View dar page (filtered by DAC) 
  * Vote as member  
  * Vote as chair 
  * Collect votes on dar 
  * Final vote on dar
